lib/ai/embeddings.go

```diff
@@ -205,11 +215,16 @@ func (e *EmbeddingProcessor) mapProcessFn(ctx context.Context, data []*nodeStrin
 }
 
 // Run runs the EmbeddingProcessor.
-func (e *EmbeddingProcessor) Run(ctx context.Context, period time.Duration) error {
+func (e *EmbeddingProcessor) Run(ctx context.Context, initialDelay, period time.Duration) error {
+	initTimer := time.NewTimer(initialDelay)
 	for {
 		select {
 		case <-ctx.Done():
 			return ctx.Err()
+		case <-initTimer.C:
+			// Stop the timer after the initial delay.
+			initTimer.Stop()
+			e.process(ctx)
 		case <-time.After(e.jitter(period)):
 			e.process(ctx)
 		}
```

Comment:
Non blocking: it looks like a great use-case for the interval util: https://github.com/gravitational/teleport/blob/master/lib/utils/interval/interval.go

Reply:
I'm not sure how the interval util could be used there. In this case, I want to run the timer once, and the interval utils only provide tools to run action periodically. I could also use even `time.Sleep` here for simplicity, except this would break the `ctx` support for the first run